### DISCUSSION TITLE
How to save the solution vector?

### DISCUSSION THREAD

**Discussion by nanguaxiaofendui** [libMesh/libmesh#3762] [Q&A]
I made a loop, how do I save each solution vector?Please help me, thank you!

code:
for(auto m : make_range(M)){  //loop
    equation_systems.init();  
    equation_systems.parameters.set<Real>("a") = m;
    equation_systems.get_system("TEST").attach_assemble_function(assemble);
    equation_systems.get_system("TEST").solve();
    NumericVector<Number> & solution = *equation_systems.get_system("TEST").solution;
}

**Comment by roystgnr** [comment]
You're taking a reference to the solution vector with `&`, not a copy.

You can't copy a `NumericVector` since it's an abstract base class, but you can call `clone()` on one to get a smart-pointer to a copy of the same sort of derived class (e.g. `PetscVector` if you've built with PETSc) as the original.

**Reply by nanguaxiaofendui** [reply]
Thank you very much!
